Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+          - matchExpressions:​
+              - key: role​
+                operator: In​
+                values:​
+                  - storage-node
+          # add affinity for nodes with MON SSDs
+          - matchExpressions:​
+              - key: ceph-mon-ssd​
+                operator: In​
+                values:​
+                  - "true"
+```
+
+We want to schedule MONs so they are spread across failure domains whenever possible. We will
+accomplish this by applying Anti-affinity between MON pods. Rook labels all MON pods
+`app=rook-ceph-mon`, and that is what will be used to spread the monitors apart. There is one rule
+for rooms, and one for racks if a room is down. We want to ensure a higher `weight` is given to
+riskier co-location scenarios.
+
+We do not recommend running MONs on the same node unless absolutely necessary. Rook automatically
+applies an Anti-affinity with medium-level `weight`. However, this might not be appropriate for all
+scenarios. For our scenario, we only want node-level co-location in the worst of failure scenarios,
+so we want to apply the highest possible `weight` of Anti-affinity for nodes.
+
+**`cluster.yaml`:**
+```yaml
+placement:
+  mon:
+    # ... nodeAffinity from above ...
+    podAntiAffinity:​
+      preferredDuringSchedulingIgnoredDuringExecution:​
+        - weight: 80​
+          podAffinityTerm:​
+            labelSelector:​
+              matchLabels:​
+                app: rook-ceph-mon​
+            topologyKey: topology.rook.io/room​
+        - weight: 90​
+          podAffinityTerm:​
+            labelSelector:​
+              matchLabels:​
+                app: rook-ceph-mon​
+            topologyKey: topology.rook.io/rack​
+        - weight: 100​
+          podAffinityTerm:​
+            labelSelector:​
+              matchLabels:​
+                app: rook-ceph-mon​
+            topologyKey: kubernetes.io/hostname​
+```
+
+> **NOTE:** If host networking is enabled, you cannot co-locate MONs because the ports will collide
+> on nodes. To enforce this, if host networking is enabled, Rook will automatically set a
+> `requiredDuringSchedulingIgnoredDuringExecution` Pod Anti-affinity rule.
+
+### Ceph OSDs
+
+There is a lot of planning that goes into the placement of monitors, and this is also true for OSDs.
+Fortunately, because the planning is already done with the monitors and because we have discussed
+the methods, it is quite a bit easier to discuss best practices for the OSDs.
+
+![Example Datacenter - OSD placement](media/ceph-bp-example-datacenter-osd-placement.png)
+
+There are two ways to select nodes to use for OSDs:
+* Apply Kubernetes Node labels and tell Rook to look for those labels. Specify in the cluster.yaml
+  `storage:useAllNodes: true` and specify `osd` `nodeAffinity` using a `ceph-osd=true` label using
+  the same Affinity methods we used for `mon`s.
+ * Specify node names in the `CephCluster` definition (`cluster.yaml`) individually in `storage:nodes`.
```

Comment:
For a best practices doc I'd suggest we focus on using the node labels instead of listing the individual nodes with this element of the CR.